<launch>
  <!-- static transforms needed for our relative odometry frames -->
  <!-- map to odom -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="odom_tf" args="0 0 0 0 0 0 1 map odom" />

  <!-- odom to rel_odom -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_odom_tf" args="0 0 0 0 0 0 1 odom rel_odom" />

  <!-- rel_odom to rel_c3po/odom-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_c3po_odom_tf" args="0 0 0 0 0 0 1 rel_odom rel_c3po/odom" />

  <!-- odom to rel_odom -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="rel_r2d2_odom_tf" args="0 0 0 0 0 0 1 rel_odom rel_r2d2/odom" />

  <!-- Run the map server -->
  <!-- Please check the file path for map files (yaml and pgm) before execution -->
  <arg name="map_file" default="$(find robotino_navigation)/maps/sample.yaml" />
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="/map" />
  </node>

  <group ns="c3po">
    <param name="tf_prefix" value="rel_c3po" />
    <!--param name="amcl/initial_pose_x" value="1" /-->
    <!--param name="amcl/initial_pose_y" value="1" /-->
    <include file="$(find tud_momana)/launch/robotino_move_base.launch" />
  </group>

  <group ns="r2d2">
    <param name="tf_prefix" value="rel_r2d2" />
    <!--param name="amcl/initial_pose_x" value="-1" /-->
    <!--param name="amcl/initial_pose_y" value="1" /-->
    <include file="$(find tud_momana)/launch/robotino_move_base.launch" />
  </group>
</launch>
